### DISCUSSION TITLE
Distributor Ring is not removing Unhealthy Pods automatically

### DISCUSSION THREAD

**Discussion by Shindek77** [grafana/tempo#3722] [Q&A]
Hello,

We have deployed tempo on our cluster in one namespace. On distributor, HPA is enabled and it's of type deployment.

We saw our traces/data was getting dropped. So while investigation we found out that  **on the distributor ring we can see the few distributor pods with state UNHEALTH and few other pods with state HEALTHY**

When we checked on cluster the pods with state HEALTHY are currently running on cluster but the pods with state UNHEALTHY re not currently running on our cluster.
Also on memberlist of tempo, we can see only HEALTHY distributor instances.
<img width="980" alt="image" src="https://github.com/grafana/tempo/assets/130061709/2b1105c3-cfd4-4621-8f33-7f8328891891">

Then as per our understanding the ring to unregister them automatically, but its not doing why????? and how to get rid of this issue???

**Comment by joe-elliott** [comment]
By default the ring does not remove unhealthy instances as it considers this to be an operational issue that must handled manually. Unhealthy distributors should not impact ingestion. The distributor ring is only used to count distributors for global rate limit enforcement.

To prevent unhealthy distributors you need to make sure they are given time to shut down cleanly and unregister themselves from the ring. [An improvement](https://github.com/grafana/tempo/pull/3687) was made here recently.
